### PR TITLE
Add a regression test following issue #4610

### LINKED ISSUE #4610: False positives `unsupported-assignment-operation` and `unsubscriptable-object` for `TypedDict`s
### Steps to reproduce

Given a file `test.py`:
```python
# pylint: disable=missing-docstring, pointless-statement
from typing import TypedDict


class MyDict(TypedDict):
    sample: int


my_dict = MyDict(sample=1)

my_dict['sample']
my_dict['sample'] = 2

```

### Current behavior

Result of `pylint test.py`:
```
************* Module test
test.py:11:0: E1136: Value 'my_dict' is unsubscriptable (unsubscriptable-object)
test.py:12:0: E1137: 'my_dict' does not support item assignment (unsupported-assignment-operation)
```

### Expected behavior

It should be passing.

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0a3
astroid 2.6.0
Python 3.9.5 (default, May 24 2021, 12:50:35) 
[GCC 11.1.0]

```

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Description

Add a regression test following issue #4610

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :scroll: Docs          |

--
Closes #4610

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
@cdce8p do you think what's wrong here ? In python 3.9 and 3.10 we have what I think are false positives, but not with python 3.8.
```python
from typing import TypedDict


class MyDict(TypedDict):
    sample: int


my_dict = MyDict(sample=1)

my_dict['sample'] # [unsubscriptable-object] False positive ?
my_dict['sample'] = 2 # [unsupported-assignment-operation] False positive ?
```

**Comment by cdce8p** [comment]
> @cdce8p do you think what's wrong here ? In python 3.9 and 3.10 we have what I think are false positives, but not with python 3.8.

Python 3.9 changed how TypedDicts are defined in the stdlib. A while back I added an inference tip for it in astroid. Unfortunately, I missed adding `dict` as base. I'm just surprised that nobody noticed it earlier 😅

[astroid -> brain_typing.py](https://github.com/PyCQA/astroid/blob/087431dc3cbb8c81c68bcbb2a2e6fcd0ec22556a/astroid/brain/brain_typing.py#L194-L204)

I'll prepare a MR to fix it shortly. Would you mind moving the regression test to `functional/t/typedDict.py`?

File: tests/functional/t/typedDict.py [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
 class CustomTD3(CustomTD2):
     var3: int
 
+
 CustomTD4 = TypedDict("CustomTD4", var4=bool)
 
 CustomTD5 = TypedDict("CustomTD5", {"var5": bool})
+
+# Checks that class used in a subscript supports subscription
+# (i.e. defines __class_getitem__ method).
```
Comment (cdce8p) [inline tests/functional/t/typedDict.py:24]: Did you mean `__getitem__`? AFAIK `__class_getitem__` is mostly used for typing purposes at the moment.
I would recommend to just remove it as it's not really necessary.

File: tests/functional/t/typedDict.py [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+
+
+class MyDict(TypedDict):
+    sample: int
```
Comment (cdce8p) [inline tests/functional/t/typedDict.py:28]: You could use `CustomTD` for it

**Review by cdce8p — COMMENTED** [review]
Just some small comments

**Review by cdce8p — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
@cdce8p do you have any idea why we're hitting a ``Failed to restore Python venv from cache`` with python 3.10 on windows ?

**Comment by cdce8p** [comment]
> @cdce8p do you have any idea why we're hitting a `Failed to restore Python venv from cache` with python 3.10 on windows ?

@Pierre-Sassoulas I'll take a look. At first glance there is nothing obviously wrong with it.

**Comment by cdce8p** [comment]
It seems to work now after I bumped the cache version.

**Comment by Pierre-Sassoulas** [comment]
Very strange indeed. Thank you for fixing this 😄

**Pull request merged**